libs/wallet/src/provider.tsx

```diff
@@ -30,29 +30,38 @@ export const VegaWalletProvider = ({ children }: VegaWalletProviderProps) => {
   // Reference to the current connector instance
   const connector = useRef<VegaConnector | null>(null);
 
-  const connect = useCallback(async (c: VegaConnector) => {
-    connector.current = c;
-    try {
-      const res = await connector.current.connect();
-
-      if (!res) {
+  const connect = useCallback(
+    async (c: VegaConnector) => {
+      connector.current = c;
+      try {
+        const res = await connector.current.connect();
+
+        if (!res) {
+          return null;
+        }
+
+        const publicKeysWithName = res.map((pk) => {
+          const nameMeta = pk.meta?.find((m) => m.key === 'name');
+          return {
+            ...pk,
+            name: nameMeta?.value ? nameMeta.value : 'None',
```

Comment:
chore (non-blocking): should show truncated key name if no name is set

Reply:
I'm not sure on this one. I feel like name should be left null otherwise it could lead to some inconsistent looking UIs if you were to say loop over all keys and render out the name?